crates/bevy_winit/src/lib.rs

```diff
@@ -29,7 +29,56 @@ impl Plugin for WinitPlugin {
             // stopping us. there are plans to remove the lifetime: https://github.com/rust-windowing/winit/pull/1456
             // .add_event::<winit::event::WindowEvent>()
             .init_resource::<WinitWindows>()
-            .set_runner(winit_runner);
+            .set_runner(winit_runner)
+            .add_system(change_window.system());
+    }
+}
+
+fn change_window(winit_windows: Res<WinitWindows>, mut windows: ResMut<Windows>) {
+    for bevy_window in windows.iter_mut() {
+        let id = bevy_window.id();
+        let window_height = bevy_window.height();
+        let window_width = bevy_window.width();
```

Comment:
the width/height variables are only used by the `bevy_window::WindowMode::Fullscreen { use_size }` match arm. Normally I would ask that we put them inside that match arm, but in this case I think we need to encode the current width/height inside of the SetWindowMode command to ensure we use the correct resolution from the point in time where the command was generated. Ex: currently if someone calls:

```
1. SetResolution w: 300 h: 500
2. SetWindowMode: Fullscreen
3. SetResolution: w: 100 h: 200
```

We end up using 100, 200 in the SetWindowMode instead of 300, 500

Reply:
I ended up retrieving the size from the `winit` window.

I was not happy with putting the size in the `SetWindowMode` command, as it would have been only useful with `Fullscreen` with `use_size` at `true`, and forced to otherwise update the window size just in case the user would have sent something different.

Let me know if that's not okay

Reply:
I think I'd still prefer to solve the "bug" above, even at the expense of clarity. How about we make it Option<(usize, usize)>, then only set it to Some() for SetWindowMode::Fullscreen